### DISCUSSION TITLE
Relative/dynamic sizing/width?

### DISCUSSION THREAD

**Discussion by luap2703** [wojtekmaj/react-pdf#1467] [Q&A]
Hi and thank you for the great library.

This seems to be a ridiculously stupid question but is there no way of **assigning 100% width to the pages?** i would like to present the pdfs in an MUI Dialog which has different sizes on different devices.

Thanks for answering!

**Comment by wojtekmaj** [comment]
If you want to make page width relative to window size, you will need to listen to `window` `resize` event and update width, in pixels, using JS. You may use [`useWindowWidth` hook](https://github.com/wojtekmaj/react-hooks#useWindowWidth) to make it easier to handle.

If you want to make page width relative to another container size, you will need to use `ResizeObserver` in a similar fashion. There's also a hook for that - [`useResizeObserver`](https://github.com/wojtekmaj/react-hooks#useResizeObserver) :)
